esp/services/espcontrol/ws_espcontrolservice.cpp

```diff
@@ -75,6 +75,104 @@ void CWSESPControlEx::init(IPropertyTree *cfg, const char *process, const char *
     }
 }
 
+bool CWSESPControlEx::onDetachFromDali(IEspContext& context, IEspDetachFromDaliRequest& req, IEspDetachFromDaliResponse& resp)
+{
+    int status = 0;
+    bool force = req.getForce_isNull() ? false : req.getForce();
+    VStringBuffer message("Request to DETACH ESP Process '%s' from Dali ", this->espProcess.get());
+    CEspServer * thisESPServer = static_cast<CEspServer *>(m_container);
+    if (thisESPServer)
+    {
+        message.append("has been issued - ");
+        if (thisESPServer->detachESPFromDali(force))
+            message.append("and success reported.");
+        else
+        {
+            message.append("but failure reported.");
+            status = -1;
+        }
+    }
+    else
+    {
+	message.append("COULD NOT BE ISSUED due to internal error");
+	status = -1;
+    }
+    resp.setMessage(message.str());
+    resp.setStatus(status);
+    return status;
+}
+
+bool CWSESPControlEx::onAttachToDali(IEspContext& context, IEspAttachToDaliRequest& req, IEspAttachToDaliResponse& resp)
+{
```

Comment:
Looks like you could write a common helper function to common up most of the code in onAttachToDali and onDetachFromDali?

Reply:
@rpastrana ?

Reply:
Take a look at handleDaliAttachmentRequest, if that's not what you intended, please let me know what you are envisioning.